### PR TITLE
1.3-atm summital

### PR DESCRIPTION
**Pull request by seanmpage**

### CODE REVIEW

**Comment by rpearce** [comment]
I can't even handle this gif

File: assignment.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    valid_user = user    
+    end
+  end
+  valid_user
+end
+
+#Prompting customer for name and loops til a name is entered
+name_loop = 0
+while name_loop == 0 
+  puts "Please enter your name >> "
```
Comment (rpearce) [inline assignment.rb]: Did you mean for the input to be on a new line? If not, use `print`

**Comment by rpearce** [comment]
There's a bug in this code (Stanley had the same one!) where, if I put in an incorrect name, I can't ever get back to enter a new name

**Comment by rpearce** [comment]
Don't worry about it for this one, but I would keep the user in the program (while...) when something is incorrect or too much money is asked to be removed from your account or the ATM (i.e., send them back to the menu screen or something)

File: assignment.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+all_users.push(user_one, user_two, user_three)
+all_questions = ["1 - Check Balance", "2 - Withdraw Money", "3 - Cancel"]
+
+atm_amount = 10000
+
+def validate_user(all_users, name, pin)
+  valid_user = nil
+  all_users.each do |user|
+    if user[:name] == name && user[:pin] == pin
+    valid_user = user
```
Comment (rpearce) [inline assignment.rb]: don't forget to indent for `if` blocks!

File: assignment.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    else
+      name_loop = 1
+    end
+end 
+#Prompting customer for pin and loops til you enter proper pin
+pin_loop = 0
+while pin_loop == 0
+  puts "Please enter your pin >> "
+  entered_pin = gets.chomp.to_i
+    if entered_pin == 0
```
Comment (rpearce) [inline assignment.rb]: this `if` block shouldn't be indented like this

File: assignment.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    else
+      pin_loop = 1
+    end
+end
+
+current_user = validate_user(all_users, entered_name, entered_pin)
+
+if current_user
+  input = 0
+  until input.between?(1,3)
```
Comment (rpearce) [inline assignment.rb]: fancy!

File: assignment.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  when 2
+    puts "How much do you want to withdraw?"
+    withdraw_amount = gets.chomp.to_i
+    
+    if withdraw_amount > atm_amount
+      puts "Atm machine does not have that amount, sorry"
+    elsif withdraw_amount > current_user[:balance]
+      puts "You broke!"
+    else
+      current_user[:balance] -= withdraw_amount
```
Comment (rpearce) [inline assignment.rb]: might want to add an `if` statement to make sure the requested amount is `> 0`!

**Comment by seanmpage** [comment]
Haha, yeah after I posted this gif....i was like maybe I should choose a different one, it gets weirder and weirder the more you look at it. I was really trying to make this code just functional when I knew the next assignment would cover a lot of what would fix in this assignment. Appreciate your input and time!

**Pull request merged**